invesalius/gui/widgets/canvas_renderer.py

```diff
@@ -67,59 +100,59 @@ def __init__(self, viewer, evt_renderer, canvas_renderer, orientation=None):
         self.canvas_renderer = canvas_renderer
         self.evt_renderer = evt_renderer
         self._size = self.canvas_renderer.GetSize()
-        self.draw_list = []
-        self._ordered_draw_list = []
+        self.draw_list: List[CanvasHandlerBase] = []
+        self._ordered_draw_list: List[Tuple[int, CanvasHandlerBase]] = []
         self.orientation = orientation
-        self.gc = None
-        self.last_cam_modif_time = -1
-        self.modified = True
-        self._drawn = False
+        self.gc: Optional[wx.GraphicsContext] = None
+        self.last_cam_modif_time: int = -1
+        self.modified: bool = True
+        self._drawn: bool = False
         self._init_canvas()
 
         self._over_obj = None
         self._drag_obj = None
         self._selected_obj = None
```

Comment:
I think you can type them as CanvasObjects

Reply:
I don't think they are just CanvasObjects as these lines exist:

```python
was_over = self._over_obj
...
was_over.on_mouse_leave(evt_obj)
```
```python
self._over_obj.on_mouse_enter(evt_obj)
```
self._selected_obj.on_deselect(evt_obj)
```

except if I increase these methods in the `CanvasObjects` Protocol

Reply:
I think it's not needed because the calling of `on_mouse_leave` and `on_deselect` are inside a `try ... except AttributeError`. This is used to try to call `on_mouse_leave`  `on_deselect`. The idea is if a object implements a callback to a event the callback is called.

Reply:
You are correct, However static type checkers would think that is not the best way to handle it and would always lint it as an error. I think we can just ignore it for now since this file is very dense and needs more time.

Reply:
Ok. We can ignore this now.